activation/nipost_test.go

```diff
@@ -807,6 +807,320 @@ func TestNIPSTBuilder_PoetUnstable(t *testing.T) {
 	})
 }
 
+// TestNIPoSTBuilder_PoETConfigChange checks if
+// it properly detects added/deleted PoET services and re-registers if needed.
+func TestNIPoSTBuilder_PoETConfigChange(t *testing.T) {
```

Comment:
These tests could be simplified now, given the logic is moved to the `submitPoetChallenges` function. The tests could test that function directly and they wouldn't need so many mocks. The tests would only assert that the function returns the expected slice of registrations. Wdyt?

Reply:
I would keep it so, because I think tests must check interfaces methods and nor particular implementation. Then if we would like to replace current  NIPoSTBuilder with some new, we wouldn't need to reimplement all tests.

Reply:
They already assume a lot about the implementation and will most likely need to be updated each time the `BuildNipost()` code changes (look how many i/f had to be mocked).  In my opinion, these tests would be easier to follow if they tested the behavior of `submitPoetChallenges` method only.

Reply:
@ConvallariaMaj What do we do about this?

Reply:
Sorry, forgot to replay, because had other PR in review and other task. 
We can do that, but I think, that in general we have to test interfaces instead of concrete implementations. 
But since it is already testing implementations, than ok

Reply:
Sorry, forgot to replay, because had other PR in review and other task.
We can do that, but I think, that in general we have to test interfaces instead of concrete implementations.
But since it is already testing implementations, than ok